gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/ApiService_CreateWithDefinitionTest.java

```diff
@@ -216,11 +259,107 @@ public void shouldNotStartApiIfNotManagedByKubernetes() throws Exception {
 
         when(apiMetadataService.fetchMetadataForApi(any(), any())).thenReturn(new ApiEntity());
 
-        apiService.createWithApiDefinition(EXECUTION_CONTEXT, api, "", definition);
+        apiService.createWithApiDefinition(EXECUTION_CONTEXT, api, USERNAME, definition);
 
         verify(apiRepository, times(1)).create(argThat(arg -> arg.getLifecycleState().equals(LifecycleState.STOPPED)));
     }
 
+    @Test
+    public void should_not_create_if_tag_does_not_exist() throws Exception {
+        JsonNode definition = readDefinition("/io/gravitee/rest/api/management/service/import-api.definition+plans.json");
+        UpdateApiEntity api = new UpdateApiEntity();
+        Proxy proxy = new Proxy();
+        EndpointGroup endpointGroup = new EndpointGroup();
+        endpointGroup.setName("endpointGroupName");
+        Endpoint endpoint = Endpoint.builder().name("endpointName").build();
+        endpointGroup.setEndpoints(singleton(endpoint));
+        proxy.setGroups(singleton(endpointGroup));
+        proxy.setVirtualHosts(Collections.singletonList(new VirtualHost("/context")));
+        api.setProxy(proxy);
+        api.setVersion("1.0");
+        api.setName("tag test");
+        api.setDescription("tag test example");
+        api.setTags(Set.of("unit-tests"));
+
+        when(tagService.findByIdAndReference("unit-tests", GraviteeContext.getCurrentEnvironment(), TagReferenceType.ORGANIZATION)).thenThrow(new TagNotFoundException("unit-tests"));
+
+        try {
+            apiService.createWithApiDefinition(EXECUTION_CONTEXT, api, USERNAME, definition);
+
+        } catch (Exception e) {
+            assertEquals(TagNotFoundException.class, e.getClass());
+            verify(apiRepository, never()).create(any());
+            return;
+        }
+        fail("Should have thrown TagNotFoundException");
+    }
```

Comment:
FYI you can rely on AssertJ assertion to prevent manually try/catch error: 

```suggestion
assertThatThrownBy(() -> {
  apiService.createWithApiDefinition(EXECUTION_CONTEXT, api, USERNAME, definition);		
}).isInstanceOf(TagNotFoundException.class);
  
verify(apiRepository, never()).create(any());
  
```